python/ray/serve/schema.py

```diff
@@ -509,7 +508,7 @@ def runtime_env_contains_remote_uris(cls, v):
             return
 
         uris = v.get("py_modules", [])
-        if "working_dir" in v:
+        if "working_dir" in v and v["working_dir"] not in uris:
```

Comment:
Non-blocker: to ensure there is no duplicate of any uris, probably best to cast to set then cast back to list before returning 🙃